instance/models/instance.py

```diff
@@ -66,13 +68,114 @@ class InconsistentInstanceState(Exception):
     """
     pass
 
+
+# States ######################################################################
+
+class InstanceState(ResourceState):
+    """
+    A [finite state machine] state describing an instance.
+    """
+    # A state is "steady" if we don't expect it to change.
+    # This information can be used to:
+    # - delay execution of an operation until the target server reaches a steady state
+    # - raise an exception when trying to schedule an operation that depends on a state change
+    #   while the target server is in a steady state
+    # Steady states include:
+    # - Status.New
+    # - Status.Running
+    # - Status.ConfigurationFailed
+    # - Status.Error
+    # - Status.Terminated
+    is_steady_state = True
+
+    # An instance is healthy if it is part of a normal (expected) workflow.
+    # This information can be used to delay execution of an operation
+    # until the target server has reached a healthy state.
+    # Healthy states include:
+    # - Status.New
+    # - Status.WaitingForServer
+    # - Status.ConfiguringServer
+    # - Status.Running
+    # - Status.Terminated
+    is_healthy_state = True
+
+
+class Status(ResourceState.Enum):
+    """
+    The states that an instance can be in.
+    """
+
+    class New(InstanceState):
+        """ Newly created """
+        state_id = 'new'
+
+    class WaitingForServer(InstanceState):
+        """ Server not yet accessible """
```

Comment:
This status class should also define `name = "Waiting for Server"`. Same with `ConfiguringServer` and `ConfigurationFailed`.

Otherwise, we see class names in the UI, instead of names with a space in them:
![screen shot 2016-04-26 at 9 42 50 pm](https://cloud.githubusercontent.com/assets/945577/14841967/eb61d5c4-0bf7-11e6-9f48-e1f42dea227f.png)